tests/components/demo/test_geo_location.py

```diff
@@ -33,54 +28,40 @@ def mock_legacy_time(legacy_patchable_time):
     yield
 
 
-class TestDemoPlatform(unittest.TestCase):
-    """Test the demo platform."""
-
-    def setUp(self):
-        """Initialize values for this testcase class."""
-        self.hass = get_test_home_assistant()
-        self.addCleanup(self.hass.stop)
-
-    def test_setup_platform(self):
-        """Test setup of demo platform via configuration."""
-        utcnow = dt_util.utcnow()
-        # Patching 'utcnow' to gain more control over the timed update.
-        with patch("homeassistant.util.dt.utcnow", return_value=utcnow):
-            with assert_setup_component(1, geo_location.DOMAIN):
-                assert setup_component(self.hass, geo_location.DOMAIN, CONFIG)
-            self.hass.block_till_done()
+async def test_setup_platform(hass):
+    """Test setup of demo platform via configuration."""
+    utcnow = dt_util.utcnow()
+    # Patching 'utcnow' to gain more control over the timed update.
+    with patch("homeassistant.util.dt.utcnow", return_value=utcnow):
+        with assert_setup_component(1, geo_location.DOMAIN):
+            assert await async_setup_component(hass, geo_location.DOMAIN, CONFIG)
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6356951)